src/script/index.js

```diff
@@ -24,10 +26,58 @@ for (var i = 0; i < paramPairsLength; i++) {
 delete i
 delete paramPair
 
+function add_instance(text, disabled, selected) {
+    const opt = document.createElement('option');
+    opt.innerText = text;
+    if (disabled) {
+        opt.setAttribute('disabled', true);
+    }
+    if (selected) {
+        opt.setAttribute('selected', true);
+    }
+    choose_instance.appendChild(opt);
+}
+
 if (prefillInstance != null) {
-    document.getElementById('instance').value = normalizeUrl(prefillInstance);
+    const url = normalizeUrl(prefillInstance);
+    instance.value = url;
+    add_instance(url.slice(8, url.length - 1), false, true);
 }
 
+choose_instance.addEventListener('focus', function (e) {
+    if (choose_instance.children.length < 3) {
+        if (choose_instance.children.length > 1) {
+            choose_instance.children[1].remove();
+        }
+        add_instance("... loading ...", true);
+        fetch('https://api.joinmastodon.org/servers')
+          .then(response => {
+              if (!response.ok) {
+                  throw new Error('Failure response from joinmastodon.');
+              }
+              return response.json();
+          })
+          .then(servers => {
+              choose_instance.children[1].remove();
+              const domains = servers.map(obj => obj.domain);
+              domains.sort();
+              for (const domain of domains) {
+                  add_instance(domain);
+              }
+          })
+          .catch(error => {
+              choose_instance.children[1].remove();
+              add_instance("--LOADING FAILED--", true);
+              console.error(
+                'Failed to fetch servers list from joinmastodon.', error);
+          });
+    }
+})
+
+choose_instance.addEventListener('change', function (e) {
+    instance.value = `https://${choose_instance.value}`;
```

Comment:
Use the function that is already there :)


```suggestion
    instance.value = normalizeUrl(choose_instance.value);
```